neuraxle/metaopt/auto_ml.py

```diff
@@ -0,0 +1,358 @@
+        Save hyperparams, and score for a trial.
+
+        :return: (hyperparams, scores)
+        """
+        pass
+
+    def get_best_hyperparams(self, higher_score_is_better=True) -> HyperparameterSamples:
+        """
+        Get the best hyperparams from all previous trials.
+
+        :return: best hyperparams
+        :rtype: HyperparameterSamples
+        """
+        hyperparams, scores = self.load_all_trials()
+
+        best_score = None
+        best_hyperparams = None
+
+        for trial_hyperparam, trial_score in zip(hyperparams, scores):
+            if best_score is None or higher_score_is_better == (trial_score > best_score):
+                best_score = trial_score
+                best_hyperparams = trial_hyperparam
+
+        return best_hyperparams
+
+    def _get_trial_hash(self, hp_dict):
+        current_hyperparameters_hash = hashlib.md5(str.encode(str(hp_dict))).hexdigest()
+        return current_hyperparameters_hash
+
+
+class HyperparamsJSONRepository(HyperparamsRepository):
+    """
+    Hyperparams repository that saves json files for every AutoML trial.
+
+    .. seealso::
+        :class:`HyperparamsRepository`,
+        :class:`HyperparameterSamples`,
+        :class:`AutoMLSequentialWrapper`
+    """
+
+    def __init__(self, folder):
+        self.cache_folder = folder
+
+    def create_new_trial(self, hyperparams: HyperparameterSamples):
+        """
+        Create new hyperperams trial json file.
+
+        :param hyperparams: hyperparams
+        :type hyperparams: HyperparameterSamples
+        :return:
+        """
+        score = None
+        self._save_trial_json(hyperparams, score)
+
+    def load_all_trials(self) -> Tuple[List[HyperparameterSamples], List[float]]:
+        """
+        Load all hyperparameter trials with their corresponding score.
+        Reads all the saved trial json files.
+
+        :return: (hyperparams, scores)
+        """
+        all_hyperparams = []
+        all_scores = []
+
+        for base_path in glob.glob(os.path.join(self.cache_folder, '*.json')):
+            with open(base_path) as f:
+                trial_json = json.load(f)
+
+            all_hyperparams.append(HyperparameterSamples(trial_json['hyperparams']))
+            all_scores.append(trial_json['score'])
+
+        return all_hyperparams, all_scores
+
+    def save_score_for_trial(self, hyperparams: HyperparameterSamples, score: float):
+        """
+        Save hyperparams, and score for a trial.
+
+        :return: (hyperparams, scores)
+        """
+        self._save_trial_json(hyperparams, score)
+
+    def _save_trial_json(self, hyperparams, score):
+        """
+        Save trial json file.
+
+        :return: (hyperparams, scores)
+        """
+        hp_dict = hyperparams.to_flat_as_dict_primitive()
+        current_hyperparameters_hash = self._get_trial_hash(hp_dict)
+        with open(os.path.join(self.cache_folder, current_hyperparameters_hash) + '.json', 'w+') as outfile:
```

Comment:
I would like to see the score preprented before the `current_hyperparameters_hash`, like: `os.path.join(self.cache_folder, str(float(score)).replace(".", ",") + "_" + scorecurrent_hyperparameters_hash + '.json')`. Note that I included the ` + '.json'` inside the `os.path.join` rather than outside;

This way, the user can go in the folder and sort alphabetically and open the best file for easy access. This is very useful, I do this normally in other projects.